### PR TITLE
Add Funky40 support

### PR DESCRIPTION
**Pull request by TheFourthCow**
Adding files for the Funky40 a new board from /u/TheFourthCow.

## convinced keymap.c and funky40.h to work with KEYMAP changed to LAYOUT as requested.

- [ ] Core
- [X] Bugfix
- [ ] New feature
- [X] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
You do not need to keep opening new PRs. Just commit and push to the same branch and the original PR will be updated. If you've accepted suggestions on the PR, you will need to pull those changes to your local copy first.

File: keyboards/funky40/readme.mk.mk [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Funky40
+
+![Funky40](imgur link goes here when initial build is finished) 
+
+A compact 40% (12x4) ortholinear keyboard kit made and sold by /u/TheFourthCow
+
+* Keyboard Maintainer: [/u/TheFourthCow](https://github.com/TheFourthCow)
+* Hardware Supported: Funky40 with Pro Micro or Elite C
+* Hardware Availability: TBD - Currently from /u/TheFourthCow pm me on reddit!
+Make example for this keyboard (after setting up your build environment):
```
Comment (fauxpark) [inline keyboards/funky40/readme.mk.mk:10]:
```suggestion

Make example for this keyboard (after setting up your build environment):
```

File: keyboards/funky40/keymaps/default/readme.md.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+![Funky40 Layout Images](https://imgur.com/a/D13SsJB)
+
+# Default Funky40 Layout
```
Comment (fauxpark) [inline keyboards/funky40/keymaps/default/readme.md.md:3]:
```suggestion
# Default Funky40 Layout

![](https://i.imgur.com/jxoCDqx.png)
![](https://i.imgur.com/3bBL52A.png)
```
These need to be actual images. Please also remove the extra `.md` from the filename.

File: keyboards/funky40/readme.mk.mk [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Funky40
```
Comment (fauxpark) [inline keyboards/funky40/readme.mk.mk:1]: Please rename this file to `readme.md`.

File: keyboards/funky40/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ */
+
+#include "funky40.h"
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	LAYOUT(
+		KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC, 
+		KC_TAB, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, 
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_DOT, KC_DOT, KC_SLSH, KC_SFTENT, //kc_sftent is a Space Cadet mod tap code that should allow right shift to be enter when tapped right shift when held
```
Comment (fauxpark) [inline keyboards/funky40/keymaps/default/keymap.c:24]:
```suggestion
		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_DOT, KC_DOT, KC_SLSH, KC_SFTENT,
```

File: keyboards/funky40/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "funky40.h"
```
Comment (fauxpark) [inline keyboards/funky40/keymaps/default/keymap.c:17]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/funky40/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+	LAYOUT(
+		KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC, 
+		KC_TAB, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, 
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_DOT, KC_DOT, KC_SLSH, KC_SFTENT, //kc_sftent is a Space Cadet mod tap code that should allow right shift to be enter when tapped right shift when held
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, LT(1, KC_DEL), KC_SPC, KC_LEFT, KC_UP, KC_DOWN, KC_RGHT),
+										   //^This lt keycode is supposed to make this key delete when tapped and enable a momentary layer to access the next layer down when held
```
Comment (fauxpark) [inline keyboards/funky40/keymaps/default/keymap.c:26]:
```suggestion
```

File: keyboards/funky40/config.h [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x0000
```
Comment (fauxpark) [inline keyboards/funky40/config.h:22]:
```suggestion
#define PRODUCT_ID      0xC4B5
```

```
echo -n "Funky40" | shasum
c4b5678c3a7e810b1c918373932f4b285bb87b47 *-
```

File: keyboards/funky40/keymaps/default/keymap.c (hunk 5 above)
Comment (TheFourthCow) [reply keyboards/funky40/keymaps/default/keymap.c:17]: this breaks the compilation process is there a file I need to rename or download to make it work with QMK_KEYBOARD_H?

File: keyboards/funky40/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+		KC_ESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC, 
+		KC_TAB, KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, 
+<<<<<<< HEAD
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_DOT, KC_DOT, KC_SLSH, KC_SFTENT, 
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, LT(1, KC_DEL), KC_SPC, KC_LEFT, KC_UP, KC_DOWN, KC_RGHT),
+										   /
+=======
+		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_DOT, KC_DOT, KC_SLSH, KC_SFTENT,
+		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, LT(1, KC_DEL), KC_SPC, KC_LEFT, KC_UP, KC_DOWN, KC_RGHT),
+>>>>>>> 63589ee06def9d6fdfd2cde95e944df740bbce4f
```
Comment (ridingqwerty) [inline keyboards/funky40/keymaps/default/keymap.c:31]:
```suggestion
		KC_LSFT, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_DOT, KC_DOT, KC_SLSH, KC_SFTENT,
		KC_LCTL, KC_LGUI, KC_LALT, KC_SPC, LT(1, KC_DEL), KC_SPC, KC_LEFT, KC_UP, KC_DOWN, KC_RGHT),
```

Comment (TheFourthCow) [reply keyboards/funky40/keymaps/default/keymap.c]: made change, compiles correctly and should contain all requested changes!

File: keyboards/funky40/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Funky40
+
+![Funky40](imgurlinkgoesherewheninitialbuildisfinished) 
+
+A compact 40% (12x4) ortholinear keyboard kit made and sold by /u/TheFourthCow
+
+* Keyboard Maintainer: [/u/TheFourthCow](https://github.com/TheFourthCow)
+* Hardware Supported: Funky40 with Pro Micro or Elite C
+* Hardware Availability: TBD - Currently from /u/TheFourthCow pm me on reddit!
```
Comment (noroadsleft) [inline keyboards/funky40/readme.md:9]: I find this a bit cleaner.

```suggestion
* Keyboard Maintainer: [TheFourthCow](https://github.com/TheFourthCow)
* Hardware Supported: Funky40 with Pro Micro or Elite C
* Hardware Availability: TBD - Currently from /u/TheFourthCow - [pm me on reddit](https://www.reddit.com/user/TheFourthCow/)!
```

File: keyboards/funky40/readme.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+# Funky40
+
+![Funky40](imgurlinkgoesherewheninitialbuildisfinished)
```
Comment (noroadsleft) [inline keyboards/funky40/readme.md:3]: Should be able to hide this line until a photo is available.

```suggestion
<!-- ![Funky40](imgurlinkgoesherewheninitialbuildisfinished) -->
```

**Comment by fauxpark** [comment]
Just missing an info.json.

**Comment by noroadsleft** [comment]
Thanks!

I'll get the `info.json` later.

**Comment by TheFourthCow** [comment]
do I need to get the `info.json`? I'm not sure how. Thanks for reviewing my files!

**Comment by noroadsleft** [comment]
> do I need to get the `info.json`? I'm not sure how. Thanks for reviewing my files!

@TheFourthCow If you want to do it yourself, [here's how](https://docs.qmk.fm/#/reference_configurator_support). If you need any help just ping me on this PR or a new one. I'm also on [QMK Discord](https://discord.gg/Uq7gcHh), with this same username.

**Pull request merged**